utils/python/CIME/utils.py

```diff
@@ -0,0 +1,65 @@
+"""
+Common functions used by cime python scripts
+"""
+import logging
+import sys
+import os
+
+_MODEL = None
+_CIMEROOT = None
+
+def expect(condition, error_msg):
+    """
+    Similar to assert except doesn't generate an ugly stacktrace. Useful for
+    checking user error, not programming error.
+
+    >>> expect(True, "error1")
+    >>> expect(False, "error2")
+    Traceback (most recent call last):
+        ...
+    SystemExit: FAIL: error2
+    """
+    if (not condition):
+        #traceback()
+        raise SystemExit("FAIL: %s" % error_msg)
+    
+
+def get_cime_root():
+###############################################################################
+    """
+    Return the absolute path to the root of CIME that contains this script
+
+    >>> os.path.isdir(os.path.join(get_cime_root(), get_acme_scripts_location_within_cime()))
+    True
+    """
+    global _CIMEROOT
+    if (_CIMEROOT is None):
+        try:
+            _CIMEROOT = os.environ["CIMEROOT"]
+        except KeyError:
+            script_absdir = os.path.abspath(os.path.join(os.path.dirname(__file__)))
+            assert acme_script_absdir.endswith(get_python_libs_location_within_cime()), acme_script_absdir
```

Comment:
acme_script_absdir is undefined. I can worry about getting the ACME tests running if you don't have them running on hobart yet.